addons/point_of_sale/report/pos_order_report.py

```diff
@@ -66,36 +77,21 @@ def _select(self):
                 s.pricelist_id,
                 s.session_id,
                 s.account_move IS NOT NULL AS invoiced,
-                SUM(l.price_subtotal - COALESCE(l.total_cost,0) / CASE COALESCE(s.currency_rate, 0) WHEN 0 THEN 1.0 ELSE s.currency_rate END) AS margin,
+                l.price_subtotal - COALESCE(l.total_cost,0) / CASE COALESCE(s.currency_rate, 0) WHEN 0 THEN 1.0 ELSE s.currency_rate END AS margin,
                 pm.payment_method_id AS payment_method_id
         """
 
     def _from(self):
         return """
             FROM pos_order_line AS l
-                INNER JOIN pos_order s ON (s.id=l.order_id)
+                LEFT JOIN pos_order s ON (s.id=l.order_id)
```

Comment:
I understand the difference between the two joins, but I think in theory we won't have a `pos_order_line` not related to a `pos_order`. So I don't think this change will have an added value. Correct me if I'm wrong :wink:

Reply:
exactly, there is no difference here between left or inner join, because we are sure for each `order_line` there is an `order`, but i can revert this as it is the same here.